noop.go

```diff
@@ -2,8 +2,17 @@ package opentracing
 
 import "github.com/opentracing/opentracing-go/log"
 
-// A NoopTracer is a trivial implementation of Tracer for which all operations
-// are no-ops.
+// A NoopTracer is a trivial, minimum overhead implementation of Tracer
+// for which all operations are no-ops.
+//
+// The primary use of this implementation is in libraries, such as RPC
+// frameworks, that make tracing an optional feature controlled by the
+// end user. A no-op implementation allows said libraries to use it
+// as the default Tracer and to write instrumentation that does
+// not need to keep checking if the tracer instance is nil.
+//
+// For the same reason, the NoopTracer is the default "global" tracer
+// (see GlobalTracer and SetGlobalTracer functions).
```

Comment:
I added a warning below. But I don't think it's fair to say that it's not suitable for RPC frameworks - it is used in TChannel, for example. Even if we took a reverse approach and built tracing on top of distributed context propagation, there'd still be a case where API for context propagation would have a no-op implementation meaning "no context propagation", and you'd have the same problem.